pandas/core/internals/array_manager.py

```diff
@@ -1153,7 +1153,13 @@ def __init__(
     def _verify_integrity(self) -> None:
         (n_rows,) = self.shape
         assert len(self.arrays) == 1
-        assert len(self.arrays[0]) == n_rows
+        arr = self.arrays[0]
+        assert len(arr) == n_rows
+        if not arr.ndim == 1:
+            raise ValueError(
+                "Passed array should be 1-dimensional, got array with "
+                f"{arr.ndim} dimensions instead."
+            )
```

Comment:
can you share any of this with the parent classes' _verify_integrity?

Reply:
Not really, I think. The parent version assumes the object is 2D, so won't work for SingleArrayManager. 
I assume that the part of `_verify_integrity` that checks the individual array could be factored out into a shared helper function, but I would rather leave that for a follow-up.